### PR TITLE
remove ExecProbeTimeout feature gate

### PR DESCRIPTION
**Pull request by aojea**
Nothing reflected in the KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1972-kubelet-exec-probe-timeouts

The comment says remove after 1.22, so 1.27 should be safe :)

/kind cleanup
```release-note
NONE
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -277,14 +277,6 @@ const (
 	// which avoids frequent relisting of containers which helps optimize performance.
 	EventedPLEG featuregate.Feature = "EventedPLEG"
 
-	// owner: @andrewsykim @SergeyKanzhelev
-	// GA: v1.20
-	//
-	// Ensure kubelet respects exec probe timeouts. Feature gate exists in-case existing workloads
-	// may depend on old behavior where exec probe timeouts were ignored.
-	// Lock to default and remove after v1.22 based on user feedback that should be reflected in KEP #1972 update
```
Comment (liggitt) [inline pkg/features/kube_features.go:285]: I think this was asking for positive user feedback, not just relying on absence of an update... did we survey whether folks were able to enable this successfully?

**Comment by dims** [comment]
@lachie83 @jackfrancis Can you please comment?

File: pkg/features/kube_features.go (hunk 1 above)
Comment (aojea) [reply pkg/features/kube_features.go:285]: I would not expect users giving positive feedback, I expect only negative feedback in the form of bugs opened. I take  the absence of them, or at least my quick search not finding result, as positive

Comment (jackfrancis) [reply pkg/features/kube_features.go:285]: my prior tests indicated that this back-compat “fix” only aided in dockershim scenarios, I think we’re far enough past that deprecation where we can safely deprecate this feature flag

happy to add an agenda item to next week’s sig-node discussion and offer to take this back up

Comment (jackfrancis) [reply pkg/features/kube_features.go:285]: Discussed during sig-node office hours and made the following commitments:

1) I'll lead a documentation effort to describe the background of this feature gate, the behavioral differences between ExecProbeTimeout=True vs ExecProbeTimeout=False, and how to leverage the probe duration metric that landed in v1.25.0 via [104484](https://github.com/kubernetes/kubernetes/pull/104484) to assess real-world impact of removing this Feature Gate, and how to transition pod + probe implementations forward if you rely upon the old, "buggy" behaviors
2) After the documentation lands in k/k we'll do some timeboxed outreach (1-2 release cyclex?) to community to encourage folks to assess and weigh in positively or negatively to the removal of the Feature Gate
3) Pending feedback we will schedule an appropriate deprecation plan

cc @SergeyKanzhelev 

@liggitt I don't think you were in the audience, does the above sound sane?

Comment (liggitt) [reply pkg/features/kube_features.go:285]: > my prior tests indicated that this back-compat “fix” only aided in dockershim scenarios, I think we’re far enough past that deprecation where we can safely deprecate this feature flag

interesting (and surprising, given the code still exists when I thought all dockershim code had been cleaned up already)

> @liggitt I don't think you were in the audience, does the above sound sane?

yeah, that sounds very responsible, thanks

Comment (jackfrancis) [reply pkg/features/kube_features.go:285]: Well to be clear about the non-dockershim scenario (e.g., containerd) pre-v1.20: it's not that `ExecProbeTimeout=False` vs `ExecProbeTimeout=True` has has no effect, it's that the ``ExecProbeTimeout=False` configuration for non-dockershim results in a different kind of (arguably) undesirable behavior where probes _always_ succeed if the time duration exceeds the default (recall that we ship a 1 second default timeout).

That's described here:

https://github.com/kubernetes/kubernetes/pull/107480#issue-1099698155

@SergeyKanzhelev pointed out that regardless of how weird this is, it still amounts to an operational change and so we should do the due diligence of documenting all of this and reaching out to the user community.

So, in summary, once we EOL v1.23 officially we will no longer have to worry about the "customers suddenly see pods restart because probe timeout enforcement was turned on w/ a 1 second timeout", because that symptom was only ever observed for dockershim.

Hope that makes sense, it's not easy to make sense of this! :/
